src/gtest/test_checktransaction.cpp

```diff
@@ -510,5 +510,296 @@ TEST(checktransaction_tests, ScCertVersion) {
 	EXPECT_FALSE(cert.ContextualCheck(state, 219, 100));
 }
 
+TEST(TransactionManipulation, EmptyTxTransformationToMutableIsNotReversible) {
+    // CopyCtor -> CopyCtor
+    CTransaction        EmptyOriginalTx;
+    CMutableTransaction mutByCopyCtor(EmptyOriginalTx);
+    CTransaction        revertedTxByCopyCtor(mutByCopyCtor);
+
+    EXPECT_FALSE(EmptyOriginalTx == revertedTxByCopyCtor);
+    EXPECT_TRUE(EmptyOriginalTx.GetHash().IsNull());
+    EXPECT_FALSE(revertedTxByCopyCtor.GetHash().IsNull());
+
+    // AssignOp -> CopyCtor
+    CMutableTransaction mutByAssignOp;
+    mutByAssignOp = EmptyOriginalTx;
+    CTransaction revertedTxFromAssignement(mutByCopyCtor);
```

Comment:
Shouldn't be mutByAssignOp  instead of mutByCopyCtor? Or why it has been instantiated?

Reply:
You are right, it should be mutByAssignOp